Packs/IronscalesEventCollector/Integrations/IronscalesEventCollector/IronscalesEventCollector.py

```diff
@@ -0,0 +1,272 @@
+
+    def client_error_handler(self, res) -> Any:
+        try:
+            err_msg = f"Error in API call [{res.status_code}] - {res.json().get('message')}"
+            raise DemistoException(err_msg, res=res)
+        except ValueError:
+            super().client_error_handler(res)
+
+    def get_jwt_token(self, api_key: str, scopes: list) -> Dict[str, Any]:
+        try:
+            jwt_key = self._http_request(
+                method="POST",
+                url_suffix="/get-token/",
+                json_data={"key": api_key, "scopes": scopes},
+            )
+            return jwt_key["jwt"]
+        except DemistoException as e:
+            if "No company found for API key" in str(e):
+                raise DemistoException(
+                    "Authorization Error: make sure the API is set correctly"
+                )
+            raise e
+
+    def get_incident(self, incident_id: int) -> Dict[str, Any]:
+        return self._http_request(
+            method="GET",
+            url_suffix=f"/incident/{self.company_id}/details/{incident_id}",
+        )
+
+    def get_open_incident_ids(self) -> List[int]:
+        return self._http_request(
+            method="GET",
+            url_suffix=f"/incident/{self.company_id}/open/",
+        ).get("incident_ids") or []
+
+
+""" HELPER FUNCTIONS """
+
+
+def get_incident_ids_by_time(
+    client: Client,
+    incident_ids: List[int],
+    start_time: datetime,
+    start_idx: int = 0,
+    end_idx: Optional[int] = None,
+) -> List[int]:
+    """Uses binary search to determine the incident ID to start fetching from.
+    This method will be called only in the first fetch.
+
+    Args:
+        client (Client): The client object
+        incident_ids (List[int]): List of all incident IDs
+        start_time (datetime): Time to start the fetch from
+        start_idx (int): Start index for the binary search
+        end_idx (int): End index for the binary search
+
+    Returns:
+        List[int]: The list of all incident IDs to fetch.
+    """
+    if end_idx is None:
+        end_idx = len(incident_ids) - 1
+
+    current_idx = (start_idx + end_idx) // 2
+
+    incident = client.get_incident(incident_ids[current_idx])
+    incident_time = dateparser.parse(incident.get("first_reported_date", ""))
+    assert incident_time, "Missing field `first_reported_date` in incident data"
+
+    if incident_time > start_time:
+        if current_idx == start_idx:
+            return incident_ids[start_idx:]
+        return get_incident_ids_by_time(
+            client,
+            incident_ids,
+            start_time,
+            start_idx,
+            current_idx - 1,
+        )
+    if incident_time < start_time:
+        if current_idx == start_idx:
+            return incident_ids[end_idx:]
+        return get_incident_ids_by_time(
+            client,
+            incident_ids,
+            start_time,
+            current_idx + 1,
+            end_idx,
```

Comment:
Since it is not very easy to debug anyway let's add the parameter names:
```suggestion
            client,
            incident_ids,
            start_time,
            start_idx=start_idx,
            end_idx=current_idx - 1,
        )
    if incident_time < start_time:
        if current_idx == start_idx:
            return incident_ids[end_idx:]
        return get_incident_ids_by_time(
            client,
            incident_ids,
            start_time,
            start_idx=current_idx + 1,
            end_idx=end_idx,
```